### DISCUSSION TITLE
Audio doesn't swap when changing default communication devices while RustDesk is running (Windows 10)

### DISCUSSION THREAD

**Discussion by Jayinem** [rustdesk/rustdesk#9440] [General]
### Bug Description

I don't know if this is an issue on my end, a bug  or this just isn't a feature that has been implemented by RustDesk yet, but anytime I switch to another audio device (I use TV monitor + Laptop + cell phone for audio) it continues on whatever device was already set and doesn't swap. I have to restart the RustDesk session for it to switch audio. It would make RustDesk perfect for me (well that and a hotkey I could enter to swap monitors during fullscreen) if this could work for me.  

I tested it on Firefox being used on the other computer using Rustdesk, the audio source doesn't change. But when I tested it on Firefox on the computer that I'm using for RustDesk it swapped the audio. In fact it had audio coming out of my TV and Laptop speaker at the same time which I didn't know was possible because RustDesk was still using my TV and the rest of the computer was using the Laptop.

### How to Reproduce

Swap audio devices on Windows 10 while using RustDesk and let me know if it swaps the audio on RustDesk to the changed source.

### Expected Behavior

Audio should swap to the intended source like the rest of the computer. 

### Operating system(s) on local side and remote side

Windows 10 

### RustDesk Version(s) on local side and remote side

1.3.1 

### Screenshots

No screenshot would help in this situation. It's an audio issue.

**Comment by KoalaBear84** [comment]
Yes. This is what I also experience. The sound of my desktop at home is going through the work laptop speakers 🙃

**Comment by pormr** [comment]
I have to temporarily disconnect RustDesk from the Internet or restart RustDesk either on local side or remote side to get audio working again after switching default audio device. (remote side: RustDesk 1.3.5 @ Windows 11 24H2)
